### ISSUE TITLE
Memory audit: make classes disposable

### ISSUE THREAD

**Issue by evanweible-wf** [Workiva/w_transport#218]
Enable these lints:

```
- cancel_subscriptions
- close_sinks
```

Make every class disposable (add `dispose()` methods). Cancel subscriptions and close sinks.

**Comment by rmconsole-wf** [comment]
It looks like you have not included any tickets in the title of this issue. Please create a ticket for this issue, or click <a href='https://w-rmconsole.appspot.com/api/v1/issues/Workiva/w_transport/218/addTicketToTitle'>here</a> to have Rosie create one for you.

**Comment by maxwellpeterson-wf** [comment]
@evanweible-wf still work to be done on this one?

**Comment by evanweible-wf** [comment]
Yes https://github.com/Workiva/w_transport/blob/master/analysis_options.yaml#L20-L22
